cicd/internal/flags/common-flags_test.go

```diff
@@ -45,10 +45,50 @@ func TestChangedFiles(t *testing.T) {
 	}
 }
 
-func TestChangedFilesEmpty(t *testing.T) {
+func TestChangedFilesNoRegexEmpty(t *testing.T) {
 	changedFiles = ""
 	actual := ChangedFiles()
 	if len(actual) != 0 {
 		t.Errorf("Expected empty slice, but got %v of len %v", actual, len(actual))
 	}
 }
+
+func TestChangedFilesRegexes(t *testing.T) {
+	tests := []struct {
+		files    string
+		regexes  []string
+		expected []string
+	}{
+		{
```

Comment:
nit: I'd add a test case with an end of line character, which is very typical, e.g. "\.java$" regexp should match Foo.java, but not Bar.javaa.